sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/HttpParsers/HttpParsingHelper.cs

```diff
@@ -0,0 +1,226 @@
+                    results.Add(new KeyValuePair<string, string>(key, value));
+                }
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Tokenizes request path.
+        /// E.g. the string "/a/b/c/d?e=f" will be tokenized into [ "a", "b", "c", "d" ].
+        /// </summary>
+        /// <param name="requestPath">The request path.</param>
+        /// <returns>List of tokens.</returns>
+        internal static List<string> TokenizeRequestPath(string requestPath)
+        {
+            var slashPrefixShift = requestPath[0] == '/' ? 1 : 0;
+            int endIdx = requestPath.IndexOfAny(RequestPathEndDelimiters, slashPrefixShift);
+            List<string> tokens = Split(requestPath, RequestPathTokenDelimiters, slashPrefixShift, endIdx);
+
+            return tokens;
+        }
+
+        /// <summary>
+        /// Extracts parameters from query string.
+        /// </summary>
+        /// <param name="requestPath">The request path.</param>
+        /// <returns>
+        /// Dictionary of query parameters.
+        /// If parameter is specified more than once then the last value is returned.
+        /// </returns>
+        internal static Dictionary<string, string> ExtractQuryParameters(string requestPath)
+        {
+            int startIdx = requestPath.IndexOf('?') + 1;
+            if (startIdx <= 0 || startIdx >= requestPath.Length)
+            {
+                return null;
+            }
+
+            int endIdx = requestPath.IndexOf('#', startIdx);
+            if (endIdx < 0)
+            {
+                endIdx = requestPath.Length;
+            }
+
+            List<string> tokens = Split(requestPath, QueryStringTokenDelimiters, startIdx, endIdx);
+
+            int pairCount = (tokens.Count + 1) / 2;
+            if (pairCount == 0)
+            {
+                return null;
+            }
+
+            var results = new Dictionary<string, string>(pairCount);
+            for (int i = 0; i < pairCount; i++)
+            {
+                int keyIdx = 2 * i;
+                int valIdx = keyIdx + 1;
+                string key = tokens[keyIdx];
+                string value = valIdx == tokens.Count ? null : tokens[valIdx];
+                if (!string.IsNullOrEmpty(key))
+                {
+                    results[key] = value;
+                }
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Extracts the HTTP verb from dependency name.
+        /// </summary>
+        /// <param name="name">The dependency name.</param>
+        /// <param name="verb">The extracted verb (or NULL if not matched).</param>
+        /// <param name="nameWithoutVerb">The dependency name sans the extracted verb.</param>
+        /// <param name="supportedVerbs">List of supported verbs to extract.</param>
+        internal static void ExtractVerb(string name, out string verb, out string nameWithoutVerb, params string[] supportedVerbs)
```

Comment:
Would a static char trie be faster here? This looks like a prefix match then extract remainder.

I think checking the 1st character of name (G,P,O,H,D) and then directly checking the remainder of the expected string is faster.

Reply:
Removed this functionality as we decided in other conversation to remove all http parser.